[Eline Ngu Xiang Ee] iP

Elinengu opened the pull request.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Elinengu committed:
add Level-0

Elinengu committed:
add Level-1

Elinengu committed:
Level 2. Add, List

Elinengu committed:
Level 3. Mark as Done

Elinengu committed:
Add Level-4

Elinengu committed:
Add A-TextUiTesting

Elinengu committed:
Add Level-5

Elinengu committed:
add Level-6

Elinengu committed:
Add Level-7

Elinengu committed:
Merge branch 'branch-Level-7'

Elinengu committed:
Add Level-8

Elinengu committed:
Merge branch 'branch-Level-8'

Elinengu committed:
Add A-MoreOOP

Elinengu committed:
Merge branch 'A-MoreOOP'

Elinengu committed:
Add A-Packages

Elinengu committed:
Merge branch 'add-gradle-support'

Elinengu committed:
Edit A-Packages

Elinengu committed:
Add A-JUnit

Elinengu committed:
Edit A-JUnit

Elinengu committed:
Add JavaDoc comments

Elinengu committed:
Merge branch 'A-JavaDoc'

Elinengu committed:
Add A-CodingStandard

Elinengu committed:
Merge branch 'A-CodingStandard'

Elinengu committed:
Add Level-9 Find

Elinengu committed:
Merge branch 'branch-Level-9'

matchastack commented on src/main/java/elysia/Elysia.java:
```diff
@@ -0,0 +1,69 @@
+    private static ArrayList<Task> arrayLists;
+    private static String folderName = "data";
+    private static String fileName = "elysia.txt";
+    private static String filePath = "./data/elysia.txt";
```
I noticed 2 things here. 

First, `filePath` is essentially `"./" + folderName + "/" + fileName` making it redundant since you can just do the operation when required. I suggest creating a helper function `makeFilePath(String folderName, String fileName)` to do so and reuse the function for better code reusability and makes it cleaner too. 
 
Second, I see that you do not have an overloaded constructor utilising one or more of `folderName`, `fileName` and `filePath` which makes them a constant. In that case, you might consider converting to `static final` and using `UPPER_SNAKE` case variable names, ie. 
```java
private static final String FOLDER_NAME = "data";
private static final String FILE_NAME = "elysia.txt";
private static final String FILE_PATH = "./data/elysia.txt";
```
 Though I understand that you might be considering adding an overload constructor like
```java
public Elysia(String fileName) {
    // code
}
``` 
then in that case I think it is ok to leave `fileName` and but personally, I will still convert `folderName` to `private static final String FOLDER_NAME`

matchastack commented on src/main/java/elysia/Elysia.java:
```diff
@@ -0,0 +1,69 @@
+
+    /**
+     * Runs the program until termination.
+     **/
```
According to the coding standard, this should be 
```java
/**
 * Runs the program until termination.
 */
```

matchastack commented on src/main/java/elysia/Elysia.java:
```diff
@@ -0,0 +1,69 @@
+ * Initializes the application.
+ */
+public class Elysia {
+    private static ArrayList<Task> arrayLists;
```
I think `arrayList` is more appropriate than `arrayLists` as `
1. it follows the class name
2. `arrayLists` suggests it is a list of lists

matchastack commented on src/main/java/elysia/Elysia.java:
```diff
@@ -0,0 +1,69 @@
+        TaskList taskList = new TaskList();
+        Ui ui = new Ui();
+
+        ui.showWelcomeMessage();
```
You might want to consider grouping the lines by separating into variable declaration/initialisation and code logic.
ie. 
```java
arrayLists = new ArrayList<>();
Scanner scanner = new Scanner(System.in);
Storage storage = new Storage(arrayLists);
TaskList taskList = new TaskList();
Ui ui = new Ui();

Storage.createFile(folderName, fileName);
storage.scanFileContents(filePath);
ui.showWelcomeMessage();
```

this helps to make the code more readable and easier to maintain/edit in the future too! Also, when you encounter merge conflicts, a separation also makes it less confusing when resolving merge conflicts.

... I noticed the same issue at other places too

matchastack commented on src/main/java/elysia/Elysia.java:
```diff
@@ -0,0 +1,69 @@
+                    taskList.addDeadline(arrayLists, input.substring(8).trim());
+                } else {
+                    taskList.addEvent(arrayLists, input.substring(5).trim());
+                }
```
I think `line 50` can be removed since the same conditions are checked individually again inside the code block

suggestion:
```java
...
} else if (input.startsWith("unmark ")) {
        taskList.unmarkAsDone(arrayLists, input);
} else if (input.startsWith("todo")) {
        taskList.addToDos(arrayLists, input.substring(4).trim());
} else if (input.startsWith("deadline")) {
        taskList.addDeadline(arrayLists, input.substring(8).trim());
} else if (input.startsWith("event")) {
        taskList.addEvent(arrayLists, input.substring(5).trim());
} else if (input.startsWith("delete")) {
        taskList.deleteTask(arrayLists, input);
} ...
```

matchastack commented on src/main/java/elysia/Elysia.java:
```diff
@@ -0,0 +1,69 @@
+
+        ui.showWelcomeMessage();
+
+
```
extra empty line

matchastack commented on src/main/java/elysia/command/Task.java:
```diff
@@ -0,0 +1,59 @@
+
+    /**
+     * Saves the task into the txt file.
+     **/
```
... I noticed the same issue at other places too

matchastack commented on src/main/java/elysia/command/TaskList.java:
```diff
@@ -0,0 +1,110 @@
+    /**
+     * Adds ToDo task to the list.
+     **/
+    public void addToDos(ArrayList<Task> arrayLists, String s) {
```
same reasoning as `arrayLists` mentioned elsewhere, I think a more appropriate name will be `addToDo` instead given that the code logic only adds 1 task at a time.

matchastack commented on src/main/java/elysia/command/TaskList.java:
```diff
@@ -0,0 +1,110 @@
+            }
+        }
+
+        Ui ui = new Ui();
```
I notice that you are creating `new Ui()` objects in every method. You might want to consider initialising somewhere in main logic and passing it as a parameter instead. Alternatively, you can consider a static method in `class Ui` if applicable

matchastack commented on src/main/java/elysia/dateparser/DateParser.java:
```diff
@@ -0,0 +1,85 @@
+import java.time.format.DateTimeParseException;
+
+import java.util.HashMap;
+import java.util.Map;
```
Very nice grouping for import statement!

matchastack commented on src/main/java/elysia/dateparser/DateParser.java:
```diff
@@ -0,0 +1,85 @@
+                DateTimeFormatter.ofPattern("yyyy/M/d"),
+                DateTimeFormatter.ofPattern("yyyy-M-d"),
+                DateTimeFormatter.ofPattern("d-M-yyyy")
+        };
```
I think this is great code! Allows for multiple user input format and done in a very neat way.

matchastack commented on src/main/java/elysia/dateparser/DateParser.java:
```diff
@@ -0,0 +1,85 @@
+                    parsedDate = LocalDate.parse(input + " " + LocalDate.now().getYear(), formatter);
+                    break; // Exit loop if parsing is successful
+                } catch (DateTimeParseException e) {
+                    // Continue to the next format if parsing fails
```
nit: redundant but I think adding a `continue` will be better. After all, it is always good to be explicit since there might be a possibility that new developers do not know the "by default" behaviour of empty `catch` code block in java. What do you think?

matchastack commented on src/main/java/elysia/dateparser/DateParser.java:
```diff
@@ -0,0 +1,85 @@
+    public static void main(String[] args) {
+        System.out.println(parseDate("2/12/2019"));
+        System.out.println(LocalTime.parse("07:30"));
+    }
```
I'm thinking this is for testing when you were writing the code for `class DateParser`? You might want to consider removing this as having multiple `main()` methods might confuse developers as that is where developers usually look for as entry point to a program. Alternatively, you can consider a JavaDoc comment to indicate its reason for existence.

matchastack commented on src/main/java/elysia/ui/Ui.java:
```diff
@@ -0,0 +1,85 @@
+ */
+public class Ui {
+    private static String line = "____________________________________________________________";
+    private static String welcomeMessage = "Hi there! Did you miss me?\n" +
+            "Wherever you are and whenever you need,\n" +
+            "Elysia will always meet your expectations.";
+    private static String exitMessage = "Alright, this time we really have to say goodbye.\n" +
+            "Goodbye, Mei!";
```
Following the coding standard for this class, the break should be before `+`
ie. 
```java
private static String welcomeMessage = "Hi there! Did you miss me?\n"
            + "Wherever you are and whenever you need,\n"
            + "Elysia will always meet your expectations.";
```
other than that, good break point! Easy to visualise the output of the program. 

Side note: personally, I do prefer what you have done (break after `+`) as it tells me I have to continue reading next line like a "go-to-next-page" but oh well, just gotta follow the rules like prof said :)

matchastack commented on src/main/java/elysia/ui/Ui.java:
```diff
@@ -0,0 +1,85 @@
+    }
+
+    /**
+     * Displays added message
+     * @param task
```
I think there needs to be an empty line separation

matchastack commented on src/main/java/elysia/ui/Ui.java:
```diff
@@ -0,0 +1,85 @@
+    }
+
+    public void showFailMessage() {
+        System.out.println("Oh my! I'm so sorry,\n" +
+                "but it seems I'm not sure what that means.\n" +
+                "Let's figure it out together, shall we?");
```
... I noticed this elsewhere too

matchastack commented on src/test/java/elysia/storage/StorageTest.java:
```diff
@@ -0,0 +1,103 @@
+import java.time.LocalDate;
+import java.util.ArrayList;
+
+import static org.junit.jupiter.api.Assertions.*;
```
static import should go before normal import and also careful of wildcard import by IDE!

matchastack reviewed (commented):
Overall, I think the code is fairly well-written, easy to follow and understand. Variable names are also explicit and descriptive of its function.

jereeemyyyy commented on src/main/java/elysia/storage/Storage.java:
```diff
@@ -0,0 +1,161 @@
+
+    /**
+     * Writes the input from users to the txt file and rewrites the content in the file.
+     **/
```
Noticed that the the formatting of the JavaDocs is not very consistent. For example, your `handleExit` method has:
```
/**
     * Creates a txt file if it does not exist.
     * Saves all the input from list into txt file.
     *
     * @param folderName folderName in String.
     * @param fileName   fileName in String.
     * @param filePath   filePath in String.
     * @throws IOException
     */
    public void handleExit(String folderName, String fileName, String filePath) throws IOException {
    // code
   }
```
while your `append` method has:
```
/**
     * Appends the input from users to the txt file.
     **/
    private void appendToFile(String filePath, String textToAppend) throws IOException {
        // code
    }
```
Generally, for readability purposes, the one that you did for `handleExit` is preferred as it gives a more comprehensive summary on the parameters of the method! Other then that, the JavaDoc for `handleExit` is well formatted!

jereeemyyyy commented on src/test/java/elysia/dateparser/DateParserTest.java:
```diff
@@ -0,0 +1,26 @@
+    public void dateParser_parseDate_dateWithYear_success(){
+        assertEquals(LocalDate.of(2024,9,2), DateParser.parseDate("2nd Sep 2024"));
+    }
+
```
Generally well formatted!

jereeemyyyy commented on src/test/java/elysia/storage/StorageTest.java:
```diff
@@ -0,0 +1,103 @@
+import java.util.ArrayList;
+
+import static org.junit.jupiter.api.Assertions.*;
+
```
To add on to liauzhanyi, maybe to help remedy the issue of wildcard imports, and assuming you are using IntelliJ as your IDE, you can:

`Go to settings -> Editor -> Code style -> Java -> Imports`

And under the "Class count to use import with *" field, you can change the number to 10000 or any large number.
You can also change the "Names count to use static import with *" for static imports.
This will ensure that IntelliJ will not classify your imports from the same package into a wildcard import (Unless you have more than 10000 imports 🤣)

jereeemyyyy reviewed (commented):
Generally, I like the design of your code! It is very readable and easy to understand. However, one suggestion from me would be to add customized exceptions to handle cases where your user does not key in the correct format of the different commands. 

Anyways, LGTM. Just some nits here and there to fix! 😄

Elinengu committed:
Adjust OO structure

Elinengu committed:
Update README.md

Elinengu committed:
Add A-CheckStyle

Elinengu committed:
Merge tag 'A-CheckStyle'

Elinengu committed:
Merge branch 'master' of https://github.com/Elinengu/ip

Elinengu committed:
Add GUI

Elinengu committed:
Merge branch 'Level-10'

Elinengu committed:
Edit Level-10

Elinengu committed:
Add ElysiaCommand

Elinengu committed:
Edit branch-Level-10

Elinengu committed:
Add Assertions

Elinengu committed:
Merge pull request #1 from Elinengu/branch-A-Assertions

Add Assertions

Elinengu committed:
Merge branch 'master' of https://github.com/Elinengu/ip

Elinengu committed:
Update README.md

Elinengu committed:
A-Code quality: improve quality of code

Refactor the methods if necessary.

Elinengu committed:
Merge branch 'master' into branch-A-CodeQuality

Elinengu committed:
Merge pull request #3 from Elinengu/branch-A-CodeQuality

Merge Branch A-Code quality

Elinengu committed:
Deadline command: sorts the deadline task accordings to the deadline

Deadline tasks are sorted according to deadline.
So users can see which tasks have earlier deadline in the list.

Elinengu committed:
Update README.md

Elinengu committed:
Update README.md

Elinengu committed:
Add random voice lines at Elysia Command

Elinengu committed:
Add more error handling features

Use Local Date Time object instead of String for Event.
Add method to check if the start time and end time for Event are valid.
Add method to check if the deadline is valid.

Elinengu committed:
Add more Error Handling

Elinengu committed:
Make find command case-insensitive

Elinengu committed:
Update README.md

Elinengu committed:
Update README.md

Elinengu committed:
Update Javadoc

Elinengu committed:
Merge branch 'master' of https://github.com/Elinengu/ip

Elinengu committed:
Update README.md

Elinengu committed:
Fix the known bugs

Close the application automatically after the exit command.
Handle the exception where user keys in the invalid index.
Make the response more meaningful if the task list is empty.

Elinengu committed:
Update README.md

Clarify the date input for deadline and event commands.

Elinengu committed:
Make the respond more meaningful and adjust the logic of date parser

Elinengu committed:
Update README.md

Add more tips for event and deadline commands.

Elinengu committed:
Add missing Javadoc

Elinengu committed:
Update format for README.md

Elinengu committed:
Update README.md

Elinengu committed:
Update README.md

Elinengu committed:
Set the default locale to prevent error from date parser or time parser

Elinengu committed:
Add CI file

Elinengu committed:
Test Codev

Elinengu committed:
Add report to Codev in gradle.yml

Elinengu committed:
Test Codev

Elinengu committed:
Edit yml file

Elinengu committed:
Merge branch 'Testing-Codev'
